DependencyInjection/Compiler/AddProcessorsPass.php

```diff
@@ -25,16 +25,21 @@
  */
 class AddProcessorsPass implements CompilerPassInterface
 {
+    use PriorityTaggedServiceTrait;
+
     public function process(ContainerBuilder $container)
     {
         if (!$container->hasDefinition('monolog.logger')) {
             return;
         }
 
-        foreach ($container->findTaggedServiceIds('monolog.processor') as $id => $tags) {
+        // array_reverse is used because ProcessableHandlerTrait::pushProcessor prepends processors to the beginning of the stack
+        foreach (array_reverse($this->findAndSortTaggedServices('monolog.processor', $container)) as $reference) {
```

Comment:
how does this behave when we have multiple tags (for instance when registering the processor in specific channels or in specific handlers ?

Reply:
@stof Good catch, it didn't work. I've managed to fix it. Please let me know what you think of the fix. It's probably easiest to review it with [whitespace changes ignored](https://github.com/symfony/monolog-bundle/pull/485/commits/264cf5d4510096f149f0d29779c34f69ce3757c8?w=1).